File: app/src/main/java/com/raju/javabaseproject/data/source/remote/http/ConnectivityInterceptor.java
package com.raju.javabaseproject.data.source.remote.http;

import android.content.Context;

import com.raju.javabaseproject.data.utils.NetworkUtil;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;



public class ConnectivityInterceptor implements Interceptor {

    private Context context;

    public ConnectivityInterceptor(Context context) {
        this.context = context;
    }

    @Override
    public Response intercept(Chain chain) throws IOException {
        if (!NetworkUtil.isConnected(context)) {
            throw new ConnectivityException();
        }

        Request.Builder builder = chain.request().newBuilder();
        return chain.proceed(builder.build());
    }
}

File: app/src/main/java/com/raju/javabaseproject/data/source/remote/http/ConnectivityInterceptor.kt
package com.raju.javabaseproject.data.source.remote.http

import android.content.Context

import com.raju.javabaseproject.data.utils.NetworkUtil

import java.io.IOException

import okhttp3.Interceptor
import okhttp3.Request
import okhttp3.Response


class ConnectivityInterceptor(private val context: Context) : Interceptor {

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        if (!NetworkUtil.isConnected(context)) {
            throw ConnectivityException()
        }

        val builder = chain.request().newBuilder()
        return chain.proceed(builder.build())
    }
}

